tap_logmeinrescue/streams/base.py

```diff
@@ -191,22 +237,26 @@ def sync_data(self, parent_ids):
 
                     ctr.increment(amount=len(parsed_response['rows']))
 
+                # technician_id acts as a substream bookmark so that we
+                # can pick up in a single stream where we left off.
                 self.state = incorporate(
                     self.state, table, 'technician_id', parent_id)
-                self.state = incorporate(
-                    self.state, table, 'start_date', start_date)
-                self.state = save_state(self.state)
+                # There's no need to save `start_date` here. Even in the
+                # case of the first run the config start_date won't change
+                # so we're safe. It's acceptable to update start_date only
+                # after we're done the sync for this whole window.
+                save_state(self.state)
 
-            start_date = end_date
-            end_date = start_date + timedelta(days=7)
             technician_id = 0
 
             self.state = incorporate(
-                self.state, table, 'start_date', start_date)
+                self.state, table, 'start_date', end_date)
```

Comment:
The new `start_date` bookmark at the end of the above loop should be the `end_date` that the above loop ran with because we're now done that date window.